io/pci/distro_tools.py

```diff
@@ -30,27 +30,61 @@ class DisrtoTool(Test):
     '''
     to test different type of tool
     '''
-    @skipUnless(IS_POWER_VM,
-                "supported only on PowerVM platform")
+
     def setUp(self):
         '''
-        to check host platform
+        get all parameters
         '''
-        self.option = self.params.get("option", default='')
+        self.option = self.params.get("test_opt", default='').split(",")
+        self.tool = self.params.get("tool", default='')
 
-    def test(self):
+    @skipUnless(IS_POWER_VM,
+                "supported only on PowerVM platform")
+    def lsslot(self, option):
         '''
-        test the lsslot tool
+        run lsslot
         '''
         cmd = "lsslot"
-        if self.option == "pci":
-            cmd = "%s -d %s" % (cmd, self.option)
+        if option == "pci":
+            cmd = "%s -d %s" % (cmd, option)
         else:
-            cmd = "%s -c %s" % (cmd, self.option)
+            cmd = "%s -c %s" % (cmd, option)
         result = process.run(cmd, shell=True, ignore_status=True)
         if result.exit_status != 0:
             self.fail("Failed to display hot plug slots")
 
+    def netstat(self, option):
+        '''
+        run netstat
+        '''
+        cmd = "netstat -%s" % option
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("Failed to monitoring network connections")
+
+    def lsprop(self, option):
+        '''
+        run lsprop
+        '''
+        cmd = "lsprop -%s" % option
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("lsprop failed")
+
```

Comment:
few options support '--' and few support only '-'.
'R' --->>> supports only '-'
and rest should be used with '--'.
Please re-check

Reply:
no all option are working with '-' this.

Reply:
====>>> with -r
[root@ltczep6-lp8 ~]# lsprop -r
lsprop: invalid option -- 'r'
.bash_logout     23 20 7e 2f 2e 62 61 73 68 5f 6c 6f 67 6f 75 74#.~/.bash_logout
		 0a 0a                         

====>>>with --r

[root@ltczep6-lp8 ~]# lsprop --r
.bash_logout     23 20 7e 2f 2e 62 61 73 68 5f 6c 6f 67 6f 75 74#.~/.bash_logout
		 0a 0a  




try checking with both way and see what difference your getting

Reply:
I think it is reverse way right?
R  is the only option with -R and rest all are with --.

 if option == "R":
            cmd = "lsprop -%s" % option
        else:
            cmd = "lsprop --%s" % option

Please re-check this.